### PR TITLE
Issue create test for mindmap

### LINKED ISSUE #683: Create unit test/integration test for mindmap
## Context

We have untested code there.

## Project

File: mindmap/tools/data.js

### PR DESCRIPTION
**Pull request by owais34**
# The Pull Request is ready

- [x] fixes #683 
- [x] all actions are passing
- [x] only fixes a single issue

## Overview
Added tests using default node:test library to check : 

1. Successful run of build command.
2. Required resource files being created (.js, .css)
3. index.html contains the tags containing created resource files.

## Review points

Unit test creation not possible for every function inside mindmap/tools/data.js , since functions are not exported.

## Mindmap

- [ ] the change has been given a reason in the overview
- [ ] the change text is reasonable short

## Summary by CodeRabbit

- **New Features**
	- Introduced integration tests to enhance the reliability of the build process.
- **Tests**
	- Added new testing tools and methods to improve testing coverage and functionality.

### CODE REVIEW

File: mindmap/package.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
     "lint-fix": "eslint . --ext .js,.ts,.cjs,.json,.tsx --fix",
     "lint-styles": "stylelint src/index.css",
     "lint-css": "ilcd",
-    "knip": "knip"
+    "knip": "knip",
+    "test": "node --test"
```
Comment (Idrinth) [inline mindmap/package.json:15]: feel free to use mocha, c8 and chai like in all other parts :)

**Review by Idrinth — CHANGES_REQUESTED** [review]

File: mindmap/test/tools/data.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  describe,
+  it,
+} from 'node:test';
+import assert from 'node:assert/strict';
+
+describe('Methods defined in data.js', () => {
+  it('should be true', () => {
+    assert.equal(true, true,);
+  },);
+},);
```
[the remark this answers is not in the document]
Comment (Idrinth) [reply mindmap/test/tools/data.js:11]: it shouldn't export anything,  just  skip the file :)

File: mindmap/package.json (hunk 1 above)
Comment (owais34) [reply mindmap/package.json:15]: Understood @Idrinth ill make the requested changes :) .

File: mindmap/test/integration-test.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+const executeCommand = promisify(exec,);
+
+rmSync('cache', {
+  recursive: true,
+  force: true,
+},);
+rmSync('dist', {
+  recursive: true,
+  force: true,
+},);
```
[the remark this answers is not in the document]
Comment (Idrinth) [reply mindmap/test/integration-test.js:30]: @coderabbitai why would this change be a benefit?

File: mindmap/package.json (hunk 1 above)
Comment (Idrinth) [reply mindmap/package.json:15]: The benefit there is mostly coverage information, so we can actually see how well or badly the project is tested

**Pull request merged**